File: StriverA2Z/C++/3_ArraysQuestion/3.1_Easy/second_largest_element_in_array.cpp
// O(n)

#include<iostream>

using namespace std;

void secondLargestElement(int arr[], int n) {
    int sLargest = arr[0];
    int largest = arr[0];
    for (int i = 0; i < n; i++) {
        if (arr[i] > largest) {
            sLargest = largest;
            largest = arr[i];
        }
        if (arr[i] < largest && arr[i] > sLargest) {
            sLargest = arr[i];
        }
    }
    cout << "Second largest element is " << sLargest;
}

int main() {
    int arr[] = {1, -1, 6, 3, 4, 7};
    secondLargestElement(arr, sizeof(arr) / sizeof(arr[0]));
    return 0;
}

File: StriverA2Z/Java/Arrays/Easy/SecondLargestElementInArray.java
public class SecondLargestElementInArray {
    public static void main(String[] args) {
        int[] arr = {1, 23, 54, 8, 5, 7};
        secondLargest(arr);
    }

    static void secondLargest(int[] arr) {
        int largest = arr[0];
        int sLargest = arr[0];
        for (int j : arr) {
            if (j > largest) {
                sLargest = largest;
                largest = j;
            }
            if (j < largest && j > sLargest) {
                sLargest = j;
            }
        }
        System.out.println("Second Largest element in array is: " + sLargest);
    }
}
